### PR TITLE
Feature/refactor

### PR DESCRIPTION
**Pull request by AlaaMYahya**

### CODE REVIEW

File: src/config/data-source.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import { AddPasswordToUser1696848807636 } from "../migration/1696848807636-AddPasswordToUser";
+
+export const AppDataSource = new DataSource({
+  type: "mysql",
+  host: "localhost",
+  port: 3306,
+  username: "root",
+  password: "",
+  database: "demo",
+  entities: [User],
```
Comment (kamelth) [inline src/config/data-source.ts]: Use the `path` library to dynamically incorporate all entities from the "entities" folder. This ensures that the list doesn't require manual updates whenever new models are added.

File: src/config/data-source.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  migrations:[AddPasswordToUser1696848807636],
```
Comment (kamelth) [inline src/config/data-source.ts]: Same for the migrations

File: src/app.js
Comment (kamelth) [inline src/app.js:1]: Why we still have js files??

File: src/app.ts [hunk 3]
```diff
@@ -1,23 +1,11 @@
...
-    //   console.log("Server is running on port 3005");
-    // });
-  // })
-  
-  // .catch((error) => console.log("Error connecting to the database:", error));
+//update to  userExpressServer
+const route = express();
 
+export const app = useExpressServer(route, {
+  controllers: [UserController],
```
Comment (kamelth) [inline src/app.ts]: Use dynamic approach as well.

**Review by kamelth — CHANGES_REQUESTED** [review]

File: src/config/data-source.ts [hunk 4]
```diff
@@ -7,5 +9,8 @@ export const AppDataSource = new DataSource({
   username: "root",
   password: "",
   database: "demo",
+  entities: [path.join(__dirname, "../entities", "*.*")],
```
Comment (kamelth) [inline src/config/data-source.ts]: It's better to get **only** the js compiled file

*.js

File: src/config/data-source.ts [hunk 5]
```diff
@@ -7,5 +9,8 @@ export const AppDataSource = new DataSource({
...
+  migrations: [path.join(__dirname, "../migration", "*.*")],
```
Comment (kamelth) [inline src/config/data-source.ts]: Same here.

File: package.json [hunk 6]
```diff
@@ -30,9 +41,11 @@
     "typescript": "^5.2.2"
   },
   "dependencies": {
+    "bcrypt": "^5.1.1",
     "class-transformer": "^0.5.1",
     "class-validator": "^0.14.0",
     "express": "^4.18.2",
+    "migrations": "^1.3.0",
```
Comment (kamelth) [inline package.json:48]: Why you'r using this package?

File: package.json [hunk 7]
```diff
@@ -12,12 +12,23 @@
     "prettier:write": "prettier --write .",
     "lint": "eslint .",
     "lint:fix": "eslint --fix .",
-    "prepare": "husky install"
+    "prepare": "husky install",
+    "typeorm": "ts-node -r ./src/module-alias.ts ./node_modules/typeorm/cli",
```
Comment (kamelth) [inline package.json:16]: Not sure if the migration worked for you!
Were you able you change the DB?

**Review by kamelth — CHANGES_REQUESTED** [review]

File: package.json (hunk 6 above)
Comment (AlaaMYahya) [reply package.json:48]: because I use the command `npm i migration`

File: package.json (hunk 7 above)
Comment (AlaaMYahya) [reply package.json:16]: Yes it added a new column

File: package.json (hunk 6 above)
Comment (kamelth) [reply package.json:48]: I mean where did you use this library?

Is it because package.json scripts?

Comment (AlaaMYahya) [reply package.json:48]: Yes, I thought I had to install it to do migration.

**Review by kamelth — APPROVED** [review]

**Pull request merged**